server/index.js

```diff
@@ -6,23 +6,33 @@ const Promise = require('bluebird');
 const { readdirSync } = require('fs');
 
 const promiseQuery = Promise.promisify(db.query).bind(db);
-
-app.use(express.static(__dirname + '../../public'));
+//middleware for sending 404 for invalid listingIds
+app.use('/:listingId/', (req, res, next) => {
+  console.log('', req.params.listingId)
+  if (req.params.listingId > 99 || req.params.listingId < 0) { 
+    res.sendStatus(404);
+  }
+  next()
+})
+//serve static files at listing sub collection
+app.use('/:listingId/', express.static(__dirname + '../../public'));
 app.use(express.json());
 
+//GET API route for listing data (called once componentDidMount & getData is called)
```

Comment:
I like your notes. Makes it easy to follow your logic.

Reply:
thanks man, trying to be good about doing it.  probably helps get rid of monkey brain lol.